### ISSUE TITLE
Unexpected results when using `.match` with `Date` objects.

### ISSUE THREAD

**Issue by arnoldmanzano** [shouldjs/should.js#150]
Hi should.js team!

Noticed unexpected results when using `.match` with `Date` objects.

Answer several questions if it is a bug:

## Node.js version?
6.11.1

## Should.js version?
12.0.0
11.2.1

## How to reproduce it?
This results in a false negative fail, but I believe it should be passing.
`(new Date('2017-09-02')).should.not.match(new Date('2017-09-03'));`

This test is passing, but I believe it should be failing.
`(new Date('2017-09-02')).should.match(new Date('2017-09-03'));`

Normally, we would use `.eql` for date assertions, but it was discovered when we were matching objects with date properties. The example below is resulting in a false negative fail.
```
      ({
        a: new Date('2017-09-02'),
        b: 12
      }).should.not.match({
        a: new Date('2017-09-03'),
        b: 12
      });
```
Thanks!

**Comment by btd** [comment]
Yep this is known 'limitation'. Problem that Dates considered as empty objects. I has such bug already in past https://github.com/shouldjs/should.js/issues/42, it is literally the same bug. It seems i need to add special handling for some built-in object like Date's (maybe also for wrapper types, RegExp, Symbols).

Do you have some ideas how this formally should be handled?

**Comment by btd** [comment]
I fixed this in 13.0.0 (yes while it is a bug, it is still breaking change).
